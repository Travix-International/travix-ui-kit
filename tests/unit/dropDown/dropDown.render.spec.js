import { shallow } from 'enzyme';
import React from 'react';
import DropDown from '../../../components/dropDown/dropDown';

describe('DropDown: render', () => {
  it('should return base dropdown with required fields', () => {
    const component = shallow(
      <DropDown
        onChange={() => {}}
        options={[]}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should return base dropdown with passed string value', () => {
    const component = shallow(
      <DropDown
        onChange={() => {}}
        options={[]}
        value="testValue"
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should return base dropdown with passed number value', () => {
    const component = shallow(
      <DropDown
        onChange={() => {}}
        options={[]}
        value={1}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should return base dropdown with passed array value', () => {
    const component = shallow(
      <DropDown
        onChange={() => {}}
        options={[]}
        value={[]}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should return clearable dropdown', () => {
    const component = shallow(
      <DropDown
        clearable
        onChange={() => {}}
        options={[]}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should return multi dropdown', () => {
    const component = shallow(
      <DropDown
        multi
        onChange={() => {}}
        options={[]}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should return dropdown with passed name', () => {
    const component = shallow(
      <DropDown
        name="testName"
        onChange={() => {}}
        options={[]}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should return dropdown with shifting viewport enabled', () => {
    const component = shallow(
      <DropDown
        onChange={() => {}}
        options={[]}
        scrollMenuIntoView
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should return searchable dropdown', () => {
    const component = shallow(
      <DropDown
        onChange={() => {}}
        options={[]}
        searchable
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should return filter dropdown', () => {
    const component = shallow(
      <DropDown
        filterMode
        onChange={() => {}}
        options={[{ checked: true }]}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should return dropdown with status', () => {
    const component = shallow(
      <DropDown
        onChange={() => {}}
        options={[]}
        status="error"
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should return dropdown with icon', () => {
    const component = shallow(
      <DropDown
        icon="iconClass"
        onChange={() => {}}
        options={[]}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should return dropdown with custom class name', () => {
    const component = shallow(
      <DropDown
        className="my-custom-class"
        onChange={() => {}}
        options={[]}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should return disabled dropdown', () => {
    const component = shallow(
      <DropDown
        disabled
        onChange={() => {}}
        options={[]}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
