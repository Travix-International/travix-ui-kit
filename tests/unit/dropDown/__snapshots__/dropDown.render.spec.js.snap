// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown: render should return base dropdown with passed array value 1`] = `
<div>
  <Select
    arrowRenderer={[Function]}
    autosize={true}
    backspaceRemoves={true}
    backspaceToRemoveMessage="Press backspace to remove {label}"
    className="ui-dropdown"
    clearAllText="Clear all"
    clearRenderer={[Function]}
    clearValueText="Clear value"
    clearable={false}
    closeOnSelect={true}
    deleteRemoves={true}
    delimiter=","
    disabled={false}
    escapeClearsValue={true}
    filterOptions={[Function]}
    ignoreAccents={true}
    ignoreCase={true}
    inputProps={Object {}}
    isLoading={false}
    joinValues={false}
    labelKey="label"
    matchPos="any"
    matchProp="any"
    menuBuffer={0}
    menuRenderer={[Function]}
    multi={false}
    noResultsText="No results found"
    onBlurResetsInput={true}
    onChange={[Function]}
    onCloseResetsInput={true}
    onSelectResetsInput={true}
    openOnClick={true}
    optionComponent={[Function]}
    options={Array []}
    pageSize={5}
    placeholder=""
    removeSelected={true}
    required={false}
    rtl={false}
    scrollMenuIntoView={false}
    searchable={false}
    simpleValue={false}
    tabSelectsValue={true}
    trimFilter={true}
    value={Array []}
    valueComponent={[Function]}
    valueKey="value"
    valueRenderer={null}
  />
</div>
`;

exports[`DropDown: render should return base dropdown with passed number value 1`] = `
<div>
  <Select
    arrowRenderer={[Function]}
    autosize={true}
    backspaceRemoves={true}
    backspaceToRemoveMessage="Press backspace to remove {label}"
    className="ui-dropdown"
    clearAllText="Clear all"
    clearRenderer={[Function]}
    clearValueText="Clear value"
    clearable={false}
    closeOnSelect={true}
    deleteRemoves={true}
    delimiter=","
    disabled={false}
    escapeClearsValue={true}
    filterOptions={[Function]}
    ignoreAccents={true}
    ignoreCase={true}
    inputProps={Object {}}
    isLoading={false}
    joinValues={false}
    labelKey="label"
    matchPos="any"
    matchProp="any"
    menuBuffer={0}
    menuRenderer={[Function]}
    multi={false}
    noResultsText="No results found"
    onBlurResetsInput={true}
    onChange={[Function]}
    onCloseResetsInput={true}
    onSelectResetsInput={true}
    openOnClick={true}
    optionComponent={[Function]}
    options={Array []}
    pageSize={5}
    placeholder=""
    removeSelected={true}
    required={false}
    rtl={false}
    scrollMenuIntoView={false}
    searchable={false}
    simpleValue={false}
    tabSelectsValue={true}
    trimFilter={true}
    value={1}
    valueComponent={[Function]}
    valueKey="value"
    valueRenderer={null}
  />
</div>
`;

exports[`DropDown: render should return base dropdown with passed string value 1`] = `
<div>
  <Select
    arrowRenderer={[Function]}
    autosize={true}
    backspaceRemoves={true}
    backspaceToRemoveMessage="Press backspace to remove {label}"
    className="ui-dropdown"
    clearAllText="Clear all"
    clearRenderer={[Function]}
    clearValueText="Clear value"
    clearable={false}
    closeOnSelect={true}
    deleteRemoves={true}
    delimiter=","
    disabled={false}
    escapeClearsValue={true}
    filterOptions={[Function]}
    ignoreAccents={true}
    ignoreCase={true}
    inputProps={Object {}}
    isLoading={false}
    joinValues={false}
    labelKey="label"
    matchPos="any"
    matchProp="any"
    menuBuffer={0}
    menuRenderer={[Function]}
    multi={false}
    noResultsText="No results found"
    onBlurResetsInput={true}
    onChange={[Function]}
    onCloseResetsInput={true}
    onSelectResetsInput={true}
    openOnClick={true}
    optionComponent={[Function]}
    options={Array []}
    pageSize={5}
    placeholder=""
    removeSelected={true}
    required={false}
    rtl={false}
    scrollMenuIntoView={false}
    searchable={false}
    simpleValue={false}
    tabSelectsValue={true}
    trimFilter={true}
    value="testValue"
    valueComponent={[Function]}
    valueKey="value"
    valueRenderer={null}
  />
</div>
`;

exports[`DropDown: render should return base dropdown with required fields 1`] = `
<div>
  <Select
    arrowRenderer={[Function]}
    autosize={true}
    backspaceRemoves={true}
    backspaceToRemoveMessage="Press backspace to remove {label}"
    className="ui-dropdown"
    clearAllText="Clear all"
    clearRenderer={[Function]}
    clearValueText="Clear value"
    clearable={false}
    closeOnSelect={true}
    deleteRemoves={true}
    delimiter=","
    disabled={false}
    escapeClearsValue={true}
    filterOptions={[Function]}
    ignoreAccents={true}
    ignoreCase={true}
    inputProps={Object {}}
    isLoading={false}
    joinValues={false}
    labelKey="label"
    matchPos="any"
    matchProp="any"
    menuBuffer={0}
    menuRenderer={[Function]}
    multi={false}
    noResultsText="No results found"
    onBlurResetsInput={true}
    onChange={[Function]}
    onCloseResetsInput={true}
    onSelectResetsInput={true}
    openOnClick={true}
    optionComponent={[Function]}
    options={Array []}
    pageSize={5}
    placeholder=""
    removeSelected={true}
    required={false}
    rtl={false}
    scrollMenuIntoView={false}
    searchable={false}
    simpleValue={false}
    tabSelectsValue={true}
    trimFilter={true}
    valueComponent={[Function]}
    valueKey="value"
    valueRenderer={null}
  />
</div>
`;

exports[`DropDown: render should return clearable dropdown 1`] = `
<div>
  <Select
    arrowRenderer={[Function]}
    autosize={true}
    backspaceRemoves={true}
    backspaceToRemoveMessage="Press backspace to remove {label}"
    className="ui-dropdown"
    clearAllText="Clear all"
    clearRenderer={[Function]}
    clearValueText="Clear value"
    clearable={true}
    closeOnSelect={true}
    deleteRemoves={true}
    delimiter=","
    disabled={false}
    escapeClearsValue={true}
    filterOptions={[Function]}
    ignoreAccents={true}
    ignoreCase={true}
    inputProps={Object {}}
    isLoading={false}
    joinValues={false}
    labelKey="label"
    matchPos="any"
    matchProp="any"
    menuBuffer={0}
    menuRenderer={[Function]}
    multi={false}
    noResultsText="No results found"
    onBlurResetsInput={true}
    onChange={[Function]}
    onCloseResetsInput={true}
    onSelectResetsInput={true}
    openOnClick={true}
    optionComponent={[Function]}
    options={Array []}
    pageSize={5}
    placeholder=""
    removeSelected={true}
    required={false}
    rtl={false}
    scrollMenuIntoView={false}
    searchable={false}
    simpleValue={false}
    tabSelectsValue={true}
    trimFilter={true}
    valueComponent={[Function]}
    valueKey="value"
    valueRenderer={null}
  />
</div>
`;

exports[`DropDown: render should return disabled dropdown 1`] = `
<div>
  <Select
    arrowRenderer={[Function]}
    autosize={true}
    backspaceRemoves={true}
    backspaceToRemoveMessage="Press backspace to remove {label}"
    className="ui-dropdown"
    clearAllText="Clear all"
    clearRenderer={[Function]}
    clearValueText="Clear value"
    clearable={false}
    closeOnSelect={true}
    deleteRemoves={true}
    delimiter=","
    disabled={true}
    escapeClearsValue={true}
    filterOptions={[Function]}
    ignoreAccents={true}
    ignoreCase={true}
    inputProps={Object {}}
    isLoading={false}
    joinValues={false}
    labelKey="label"
    matchPos="any"
    matchProp="any"
    menuBuffer={0}
    menuRenderer={[Function]}
    multi={false}
    noResultsText="No results found"
    onBlurResetsInput={true}
    onChange={[Function]}
    onCloseResetsInput={true}
    onSelectResetsInput={true}
    openOnClick={true}
    optionComponent={[Function]}
    options={Array []}
    pageSize={5}
    placeholder=""
    removeSelected={true}
    required={false}
    rtl={false}
    scrollMenuIntoView={false}
    searchable={false}
    simpleValue={false}
    tabSelectsValue={true}
    trimFilter={true}
    valueComponent={[Function]}
    valueKey="value"
    valueRenderer={null}
  />
</div>
`;

exports[`DropDown: render should return dropdown with custom class name 1`] = `
<div>
  <Select
    arrowRenderer={[Function]}
    autosize={true}
    backspaceRemoves={true}
    backspaceToRemoveMessage="Press backspace to remove {label}"
    className="ui-dropdown my-custom-class"
    clearAllText="Clear all"
    clearRenderer={[Function]}
    clearValueText="Clear value"
    clearable={false}
    closeOnSelect={true}
    deleteRemoves={true}
    delimiter=","
    disabled={false}
    escapeClearsValue={true}
    filterOptions={[Function]}
    ignoreAccents={true}
    ignoreCase={true}
    inputProps={Object {}}
    isLoading={false}
    joinValues={false}
    labelKey="label"
    matchPos="any"
    matchProp="any"
    menuBuffer={0}
    menuRenderer={[Function]}
    multi={false}
    noResultsText="No results found"
    onBlurResetsInput={true}
    onChange={[Function]}
    onCloseResetsInput={true}
    onSelectResetsInput={true}
    openOnClick={true}
    optionComponent={[Function]}
    options={Array []}
    pageSize={5}
    placeholder=""
    removeSelected={true}
    required={false}
    rtl={false}
    scrollMenuIntoView={false}
    searchable={false}
    simpleValue={false}
    tabSelectsValue={true}
    trimFilter={true}
    valueComponent={[Function]}
    valueKey="value"
    valueRenderer={null}
  />
</div>
`;

exports[`DropDown: render should return dropdown with icon 1`] = `
<div>
  <Select
    arrowRenderer={[Function]}
    autosize={true}
    backspaceRemoves={true}
    backspaceToRemoveMessage="Press backspace to remove {label}"
    className="ui-dropdown"
    clearAllText="Clear all"
    clearRenderer={[Function]}
    clearValueText="Clear value"
    clearable={false}
    closeOnSelect={true}
    deleteRemoves={true}
    delimiter=","
    disabled={false}
    escapeClearsValue={true}
    filterOptions={[Function]}
    ignoreAccents={true}
    ignoreCase={true}
    inputProps={Object {}}
    isLoading={false}
    joinValues={false}
    labelKey="label"
    matchPos="any"
    matchProp="any"
    menuBuffer={0}
    menuRenderer={[Function]}
    multi={false}
    noResultsText="No results found"
    onBlurResetsInput={true}
    onChange={[Function]}
    onCloseResetsInput={true}
    onSelectResetsInput={true}
    openOnClick={true}
    optionComponent={[Function]}
    options={Array []}
    pageSize={5}
    placeholder=""
    removeSelected={true}
    required={false}
    rtl={false}
    scrollMenuIntoView={false}
    searchable={false}
    simpleValue={false}
    tabSelectsValue={true}
    trimFilter={true}
    valueComponent={[Function]}
    valueKey="value"
    valueRenderer={[Function]}
  />
</div>
`;

exports[`DropDown: render should return dropdown with passed name 1`] = `
<div>
  <Select
    arrowRenderer={[Function]}
    autosize={true}
    backspaceRemoves={true}
    backspaceToRemoveMessage="Press backspace to remove {label}"
    className="ui-dropdown"
    clearAllText="Clear all"
    clearRenderer={[Function]}
    clearValueText="Clear value"
    clearable={false}
    closeOnSelect={true}
    deleteRemoves={true}
    delimiter=","
    disabled={false}
    escapeClearsValue={true}
    filterOptions={[Function]}
    ignoreAccents={true}
    ignoreCase={true}
    inputProps={Object {}}
    isLoading={false}
    joinValues={false}
    labelKey="label"
    matchPos="any"
    matchProp="any"
    menuBuffer={0}
    menuRenderer={[Function]}
    multi={false}
    name="testName"
    noResultsText="No results found"
    onBlurResetsInput={true}
    onChange={[Function]}
    onCloseResetsInput={true}
    onSelectResetsInput={true}
    openOnClick={true}
    optionComponent={[Function]}
    options={Array []}
    pageSize={5}
    placeholder=""
    removeSelected={true}
    required={false}
    rtl={false}
    scrollMenuIntoView={false}
    searchable={false}
    simpleValue={false}
    tabSelectsValue={true}
    trimFilter={true}
    valueComponent={[Function]}
    valueKey="value"
    valueRenderer={null}
  />
</div>
`;

exports[`DropDown: render should return dropdown with shifting viewport enabled 1`] = `
<div>
  <Select
    arrowRenderer={[Function]}
    autosize={true}
    backspaceRemoves={true}
    backspaceToRemoveMessage="Press backspace to remove {label}"
    className="ui-dropdown"
    clearAllText="Clear all"
    clearRenderer={[Function]}
    clearValueText="Clear value"
    clearable={false}
    closeOnSelect={true}
    deleteRemoves={true}
    delimiter=","
    disabled={false}
    escapeClearsValue={true}
    filterOptions={[Function]}
    ignoreAccents={true}
    ignoreCase={true}
    inputProps={Object {}}
    isLoading={false}
    joinValues={false}
    labelKey="label"
    matchPos="any"
    matchProp="any"
    menuBuffer={0}
    menuRenderer={[Function]}
    multi={false}
    noResultsText="No results found"
    onBlurResetsInput={true}
    onChange={[Function]}
    onCloseResetsInput={true}
    onSelectResetsInput={true}
    openOnClick={true}
    optionComponent={[Function]}
    options={Array []}
    pageSize={5}
    placeholder=""
    removeSelected={true}
    required={false}
    rtl={false}
    scrollMenuIntoView={true}
    searchable={false}
    simpleValue={false}
    tabSelectsValue={true}
    trimFilter={true}
    valueComponent={[Function]}
    valueKey="value"
    valueRenderer={null}
  />
</div>
`;

exports[`DropDown: render should return dropdown with status 1`] = `
<div>
  <Select
    arrowRenderer={[Function]}
    autosize={true}
    backspaceRemoves={true}
    backspaceToRemoveMessage="Press backspace to remove {label}"
    className="ui-dropdown ui-dropdown_error"
    clearAllText="Clear all"
    clearRenderer={[Function]}
    clearValueText="Clear value"
    clearable={false}
    closeOnSelect={true}
    deleteRemoves={true}
    delimiter=","
    disabled={false}
    escapeClearsValue={true}
    filterOptions={[Function]}
    ignoreAccents={true}
    ignoreCase={true}
    inputProps={Object {}}
    isLoading={false}
    joinValues={false}
    labelKey="label"
    matchPos="any"
    matchProp="any"
    menuBuffer={0}
    menuRenderer={[Function]}
    multi={false}
    noResultsText="No results found"
    onBlurResetsInput={true}
    onChange={[Function]}
    onCloseResetsInput={true}
    onSelectResetsInput={true}
    openOnClick={true}
    optionComponent={[Function]}
    options={Array []}
    pageSize={5}
    placeholder=""
    removeSelected={true}
    required={false}
    rtl={false}
    scrollMenuIntoView={false}
    searchable={false}
    simpleValue={false}
    tabSelectsValue={true}
    trimFilter={true}
    valueComponent={[Function]}
    valueKey="value"
    valueRenderer={null}
  />
</div>
`;

exports[`DropDown: render should return filter dropdown 1`] = `
<div>
  <Select
    arrowRenderer={[Function]}
    autosize={true}
    backspaceRemoves={true}
    backspaceToRemoveMessage="Press backspace to remove {label}"
    className="ui-dropdown ui-dropdown_filter ui-dropdown_state-active"
    clearAllText="Clear all"
    clearRenderer={[Function]}
    clearValueText="Clear value"
    clearable={false}
    closeOnSelect={true}
    deleteRemoves={true}
    delimiter=","
    disabled={false}
    escapeClearsValue={true}
    filterOptions={[Function]}
    ignoreAccents={true}
    ignoreCase={true}
    inputProps={Object {}}
    isLoading={false}
    joinValues={false}
    labelKey="label"
    matchPos="any"
    matchProp="any"
    menuBuffer={0}
    menuRenderer={[Function]}
    multi={true}
    noResultsText="No results found"
    onBlurResetsInput={true}
    onChange={[Function]}
    onCloseResetsInput={true}
    onSelectResetsInput={true}
    openOnClick={true}
    optionComponent={[Function]}
    options={
      Array [
        Object {
          "checked": true,
        },
      ]
    }
    pageSize={5}
    placeholder=""
    removeSelected={true}
    required={false}
    rtl={false}
    scrollMenuIntoView={false}
    searchable={false}
    simpleValue={false}
    tabSelectsValue={true}
    trimFilter={true}
    valueComponent={[Function]}
    valueKey="value"
    valueRenderer={null}
  />
</div>
`;

exports[`DropDown: render should return multi dropdown 1`] = `
<div>
  <Select
    arrowRenderer={[Function]}
    autosize={true}
    backspaceRemoves={true}
    backspaceToRemoveMessage="Press backspace to remove {label}"
    className="ui-dropdown"
    clearAllText="Clear all"
    clearRenderer={[Function]}
    clearValueText="Clear value"
    clearable={false}
    closeOnSelect={true}
    deleteRemoves={true}
    delimiter=","
    disabled={false}
    escapeClearsValue={true}
    filterOptions={[Function]}
    ignoreAccents={true}
    ignoreCase={true}
    inputProps={Object {}}
    isLoading={false}
    joinValues={false}
    labelKey="label"
    matchPos="any"
    matchProp="any"
    menuBuffer={0}
    menuRenderer={[Function]}
    multi={true}
    noResultsText="No results found"
    onBlurResetsInput={true}
    onChange={[Function]}
    onCloseResetsInput={true}
    onSelectResetsInput={true}
    openOnClick={true}
    optionComponent={[Function]}
    options={Array []}
    pageSize={5}
    placeholder=""
    removeSelected={true}
    required={false}
    rtl={false}
    scrollMenuIntoView={false}
    searchable={false}
    simpleValue={false}
    tabSelectsValue={true}
    trimFilter={true}
    valueComponent={[Function]}
    valueKey="value"
    valueRenderer={null}
  />
</div>
`;

exports[`DropDown: render should return searchable dropdown 1`] = `
<div>
  <Select
    arrowRenderer={[Function]}
    autosize={true}
    backspaceRemoves={true}
    backspaceToRemoveMessage="Press backspace to remove {label}"
    className="ui-dropdown"
    clearAllText="Clear all"
    clearRenderer={[Function]}
    clearValueText="Clear value"
    clearable={false}
    closeOnSelect={true}
    deleteRemoves={true}
    delimiter=","
    disabled={false}
    escapeClearsValue={true}
    filterOptions={[Function]}
    ignoreAccents={true}
    ignoreCase={true}
    inputProps={Object {}}
    isLoading={false}
    joinValues={false}
    labelKey="label"
    matchPos="any"
    matchProp="any"
    menuBuffer={0}
    menuRenderer={[Function]}
    multi={false}
    noResultsText="No results found"
    onBlurResetsInput={true}
    onChange={[Function]}
    onCloseResetsInput={true}
    onSelectResetsInput={true}
    openOnClick={true}
    optionComponent={[Function]}
    options={Array []}
    pageSize={5}
    placeholder=""
    removeSelected={true}
    required={false}
    rtl={false}
    scrollMenuIntoView={false}
    searchable={true}
    simpleValue={false}
    tabSelectsValue={true}
    trimFilter={true}
    valueComponent={[Function]}
    valueKey="value"
    valueRenderer={null}
  />
</div>
`;
